Resources/public/js/views/universaldiscovery/ez-universaldiscoverysearchview.js

```diff
@@ -0,0 +1,500 @@
+ */
+YUI.add('ez-universaldiscoverysearchview', function (Y) {
+    "use strict";
+    /**
+     * Provides the universal discovery search method
+     *
+     * @module ez-universaldiscoverysearchview
+     */
+    Y.namespace('eZ');
+
+    var events = {
+            '.ez-ud-search-button': {
+                'tap': '_search',
+            },
+            '.ez-ud-searchresult-preview-button': {
+                'tap': '_selectContent',
+            },
+            '.ez-searchresult-navigation-link': {
+                'tap': '_handlePagination',
+            },
+        },
+        IS_SELECTED_ROW_CLASS = 'is-selected',
+        IS_PAGE_LOADING = 'is-page-loading',
+        IS_DISABLED = 'is-disabled';
+
+    function linkIsDisabled(link) {
+        return link.hasClass(IS_DISABLED);
+    }
+
+    /**
+     * The universal discovery search method view. It allows the user to pick a
+     * content from the list which is the result of the search.
+     *
+     * @namespace eZ
+     * @class UniversalDiscoverySearchView
+     * @constructor
+     * @extends eZ.UniversalDiscoveryMethodBaseView
+     */
+    Y.eZ.UniversalDiscoverySearchView = Y.Base.create('universalDiscoverySearchView',
+        Y.eZ.UniversalDiscoveryMethodBaseView, [Y.eZ.AsynchronousView], {
+        initializer: function () {
+            this._fireMethod = this._fireLocationSearch;
+            this._watchAttribute = 'searchResultList';
+
+            this.events = Y.merge(this.events, events);
+
+            this.after(['multipleChange', 'isSelectableChange'], this._setSelectedViewAttrs);
+            this.after('offsetChange', this._refresh);
+            this.after('visibleChange', this._unselectContent);
+        },
+
+        /**
+         * Custom reset implementation to explicitely reset the sub views.
+         *
+         * @method reset
+         * @param {String} [name]
+         */
+        reset: function (name) {
+            if ( name === 'selectedView' ) {
+                this.get('selectedView').reset();
+                return;
+            }
+            this.constructor.superclass.reset.apply(this, arguments);
+        },
+
+        render: function () {
+            var container = this.get('container');
+
+            container.setHTML(this.template({
+                searchText: this.get('searchText'),
+                searchResultCount: this.get('searchResultCount'),
+                searchResultList: this._convertToJSONList(),
+                isFirst: this._isFirstPage(),
+                isLast: this._isLastPage(),
+                hasPages: this._hasPages(),
+                loadingError: this.get('loadingError'),
+                multiple: this.get('multiple')
+            }));
+
+            this.get('selectedView').set('contentStruct', null);
+            container.one('.ez-ud-search-selected').append(
+                this.get('selectedView').render().get('container')
+            );
+            this._uiPageEndLoading();
```

Comment:
that's weird to that here as well. I guess this should be done in a `searchResultListChange` event handler.